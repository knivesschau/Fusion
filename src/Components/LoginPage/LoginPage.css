/* mobile styles first, min 320px */
.Login_Page {
    font-size: 1.10rem;
    text-align: center;
    border: 1px solid black;
    background-color: rgb(190, 190, 190, .8);
    border: 1px solid grey;
    border-radius: 2em;
    box-shadow: inset -2px -2px 5px 0 black;
    margin: 1.5em;
    padding: 1em;
}

.Login_Form {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin-top: -1.75em;
}

.Demo_Info {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    text-align: center;
    font-size: .95rem;
    font-weight: bold;
    margin-bottom: .5em;
    color: black;
}

#login-tag {
    font-size: 1.40rem;
    color: white; 
    text-shadow: 3px 2px 2px black;
    margin-top: .25em;
}

#return-user, #return-pass {
    font-size: 1.25rem;
    font-weight: 500;
    margin-bottom: .25em;
}

#return-app-user, #user-password { 
    padding: .25em;
    width: 70%;
}

#user-login {
    cursor: pointer;
    display: inline-block;
    box-sizing: border-box;
    background-color: #A9C09B;
    text-transform: uppercase;
    color: black;
    font-size: 1rem;
    font-weight: bold;
    margin-left: auto;
    margin-right: auto;
    margin-top: 1.25em;
    margin-bottom: .75em;
    padding: 7px 25px;
    border: 1px solid black;
    border-radius: .75em;
    transition: all .15s;
}

#user-login:hover {
    text-shadow: 0 0 32px rgba(255, 255, 255, 1);
    color: white;
    border-color: white;
}

#user-login:active {
    background-color: #881600;
}

#cooking-pot-icon {
    height: 45%;
    width: 45%;
}

/* tablet and desktop styles */
@media only screen and (min-width: 768px) {
    .Login_Page {
        margin-left: 5em;
        margin-right: 5em;
        margin-top: 3em;
        border-radius: 5em;
    }

    .Login_Form {
        margin-top: -2.5em;
    }

    .Demo_Info {
        font-size: 1.5rem;
    }

    #login-tag {
        font-size: 2.5rem;
        margin-bottom: .75em;
        margin-top: .10em;
    }

    #return-user, #return-pass, #user-login {
        font-size: 1.5rem;
    }

    #user-login {
        margin-top: .75em;
        margin-bottom: .5em;
        padding-top: .45em;
        padding-bottom: .45em;
    }

    #return-app-user, #user-password {
        width: 60%;
        font-size: 1.25rem;
    }

    #cooking-pot-icon {
        height: 30%;
        width: 30%;
    }
}

@media only screen and (min-width: 1200px) {
    .Login_Page {
        margin-left: 15em;
        margin-right: 15em;
    }

    #login-tag {
        font-size: 3.5rem;
    }

    #return-user, #return-pass, #user-login {
        font-size: 2rem;
    }

    #return-app-user, #user-password {
        width: 50%;
        font-size: 1.5rem;
    }

    #cooking-pot-icon {
        height: 25%;
        width: 25%;
    }
}

@media only screen and (min-width: 1600px) {
    .Login_Page {
        margin-left: 25em;
        margin-right: 25em;
        margin-top: 5em;
    }

    #cooking-pot-icon {
        height: 20%;
        width: 20%;
    }
} 

@media only screen and (min-width: 2000px) {
    .Login_Page {
        margin-left: 40em;
        margin-right: 40em;
    }
}