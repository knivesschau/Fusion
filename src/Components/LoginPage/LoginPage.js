import React, { Component } from 'react';
import TokenService from '../../services/token-services';
import AuthApiService from '../../services/auth-api-service';
import ErrorValidation from '../../ErrorHandlers/ErrorValidation';
import './LoginPage.css';

export default class LoginPage extends Component {
    constructor(props) {
        super(props);
        this.state = {error: null};
    };
    
    static defaultProps = {
        onValidLogin: () => {}
    };

    // handle POST request on client for user logins
    handleJwtLoginAuth = e => {
        e.preventDefault();

        const {return_username, return_password} = e.target;

        this.setState({
            error: null
        });

        AuthApiService.postLogin({
            user_name: return_username.value,
            password: return_password.value
        })
            .then(res => {
                return_username.value = '';
                return_password.value = '';
                TokenService.saveAuthToken(res.authToken);
                this.props.onValidLogin();
            })
            .then(() => {
                window.location="/your-cookbook";
            })
            .catch(res => {
                this.setState({
                    error: "Invalid username or password. Please double-check your crendentials and try again."
                });
            });
    };
    
    render() {
        return (
            <section className="Login_Page">
                
                <h3>Log In</h3>

                <form className="Login_Form" onSubmit={this.handleJwtLoginAuth}>
                    
                    <label htmlFor="username">
                        <p id="return-user">Username:</p>
                        <input type="text" id="return-app-user" name="return_username"/>
                    </label>

                    <label htmlFor="password">
                        <p id="return-pass">Password:</p>
                        <input type="password" id="user-password" name="return_password"/>
                    </label>

                    <button type="submit" id="user-login">Login</button>

                </form>

                <ErrorValidation id="login-error" message={this.state.error}/>

            </section>
        );
    }
}