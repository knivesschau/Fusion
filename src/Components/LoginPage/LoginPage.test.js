import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginPage from './LoginPage';

describe ('LoginPage Component', () => {
    it ('Renders the complete LoginPage Component and all nested Components', () => {
        const wrapper = shallow(<LoginPage/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});