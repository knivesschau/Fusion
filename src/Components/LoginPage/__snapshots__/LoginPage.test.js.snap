// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Component Renders the complete LoginPage Component and all nested Components 1`] = `
<section
  className="Login_Page"
>
  <h3>
    Log In (Will Be Included in Final Version)
  </h3>
  <form
    className="Login_Form"
  >
    <label
      htmlFor="username"
    >
      <p
        id="return-user"
      >
        Username:
      </p>
      <input
        id="return-app-user"
        name="return_username"
        type="text"
      />
    </label>
    <label
      htmlFor="password"
    >
      <p
        id="return-pass"
      >
        Password:
      </p>
      <input
        id="user-password"
        name="return_password"
        type="password"
      />
    </label>
    <button
      id="user-login"
      type="submit"
    >
      Login
    </button>
  </form>
</section>
`;
